Feature/3578 rally plugin

lmontopo opened the pull request:
Implement CAAgileCentralIntegrationPlugin.

geoffwhittington commented on sdetools/alm_integration/ca_agile_central_integration_plugin.py:
```diff
@@ -0,0 +1,80 @@
+        self.workspace = IntegrationField(
+            name='rally_workspace',
+            label='CA Agile Central Workspace',
+            placeholder='My Workspace'
```
I wouldn't put a placeholder for this as its required and no good default can be provided.

sibartlett replied:
Placeholder != default

A placeholder is example input. Although, I'm not sure "My Workspace" is a useful example?

lmontopo replied:
That is the current placeholder on the form, which is why I put it there.  I'm open to removing it and/or putting something more useful there. 

Thanks for reminding me that this is a required field!  I will add that!

geoffwhittington replied:
Usually we set Placeholders for the default value we have in sdetools commands. I just assumed it was communicating the same thing.

I've seen IE do terrible things in the past with Placeholders - making them the default value.

sibartlett replied:
IE didn't get support for placeholders until version 10.

We've been using some flaky hack to add "placeholder" support to old versions of browsers that don't support them - hence the IE issues you describe.

lmontopo commented:
Question @sibartlett @geoffwhittington : Its wasn't clear to me which things should reference 'rally' and which things should reference 'ca agile central'.  I kept the UI things consistent with what was already there (so ca agile central) and i left the id 'rally' because thats our current id.  But like, the file name for instance? And just variables in the code.  Should they consistently be 'rally' or should we try to use 'ca agile central' everywhere except for the id? Let me know what your thoughts are on this.

sibartlett commented:
I say we use "rally" everywhere that's not visible by user, just because it's nice and short. The UI must always say "CA Agile Central" though, because that's the new product name.

geoffwhittington commented on sdetools/alm_integration/alm_integration_plugin_base.py:
```diff
@@ -315,6 +315,82 @@ def __init__(self):
+            description=(
+                "If you have customized your ALM you may need to fill in the "
+                "following field. Please get in touch with SD Elements support"
+                " for help.\n{0}".format(self.DYNAMIC_MACROS))
```
We could remove the `{0}` and the `DYNAMIC_MACROS` - we could only show a couple here and they'd need to be in context.

lmontopo replied:
not sure i understand what you mean exactly.  Why would we not show all of the options?

lmontopo replied:
Ok so after talking to Simon I think that you are just referring to maybe not having this huge description here, but instead maybe listing this info in a Zendesk doc or some other way.  This is something we've discussed and that we will consider separately.  It is outside of the scope of this ticket.

geoffwhittington commented on sdetools/alm_integration/ca_agile_central_integration_plugin.py:
```diff
@@ -0,0 +1,81 @@
+            name='rally_card_type',
+            label='Issue Type',
+            placeholder='Story',
+            cli_description='IDs for issues raised in Rally'
```
Description should read, `The type of issue to create in the CA Agile Central project`

lmontopo replied:
OK. The `cli_description`'s that I've put in this document all came from what currently is the text on the command line.  I'm not against updating these strings, but just so you know that's why they are what they currently are.

geoffwhittington commented on sdetools/alm_integration/ca_agile_central_integration_plugin.py:
```diff
@@ -0,0 +1,81 @@
+        self.rally_new_statuses = IntegrationField(
+            name='rally_new_status',
+            label='Open an issue in CA Agile Central with the following state',
+            cli_description='status to set for new tasks in Rally',
```
`Status to assign to new issues in CA Agile Central`

lmontopo replied:
Cool.  The command line currently refers to the alm as 'rally' but I can update this to be CA Agile Central.

lmontopo commented:
Fixed all the changes except the DYNAMIC MACROS one and rebased so that I'm on top of Jordan's i18n changes, and I internationalized the few strings I added.

Gee19 commented on sdetools/alm_integration/alm_integration_plugin_base.py:
```diff
@@ -317,6 +317,80 @@ def __init__(self):
+            label=_('ALM Title Format'),
+            field_type=FieldType.SELECT,
+            description=_("Customize the issue titles created in the "
+                          " ALM by choosing one of the templates below."),
```
extra space here

Gee19 commented:
+1 even though I made some very minor changes to this branch :)

sibartlett commented:
+1
